core/src/main/scala/org/apache/spark/scheduler/bus/ListenerBusQueueImpl.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler.bus
+
+import com.codahale.metrics.Timer
+import scala.reflect.ClassTag
+
+import org.apache.spark.scheduler.{SparkListenerEvent, SparkListenerEventDispatcher, SparkListenerInterface}
+import org.apache.spark.scheduler.bus.ListenerBusQueue.{FixGroupOfListener, ModifiableGroupOfListener}
+
+// For generic message processor (like event logging)
+private[scheduler] class ProcessorListenerBusQueue(
```

Comment:
For the name of the file, I can change it ! Do you have a better name ? I can even put the content of the file (the 2 concrete implementations)  in the BusQueue.scala file.

I refactored a bit this file. Now I have only 2 implementations: 
 - `ProcessorBusQueue`: This is the implementation for generic processor (In which we do not do the dispatch by  event type)
-  `ListenerBusQueue`: This is the implementation for listener (with the dispatch by event type)